File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
//prompt user for how much change is owed
    float change = get_float("Change owed: ");
//makes sure that the user is re-prompted contunually if they input a negative value 
    while (change<0)
    {
        change = get_float("Change owed: ");
    }
//convert the user's inputted dollars to cents for precision
    int cents = round(change * 100);
//set number of coins equal to zero to start
    int coins = 0;   
//When quarters can be used (there is more than 25 cents of change left), decrease amount by quarter and then increase the number of coins used by 1
    while (cents >= 25)
    {
        cents = cents-25;
        coins++;
    } 
//When dimes can be used (there is more than 10 cents of change left), decrease amount by dime and then increase the number of coins used by 1
    while (cents >= 10)
    {
         cents = cents-10;
         coins++;
    }      
//When nickels can be used (there is more than 5 cents of change left), decrease amount by nickel and then increase the number of coins used by 1
    while (cents >= 5)
    {
       cents = cents-5;
       coins++;
    }   
//When pennies can be used (there is more than 1 cent of change left), decrease amount by penny and then increase the number of coins used by 1
     while (cents >= 1)
     {
          cents = cents-1;
          coins++;
     }   
//print the final value of coins used and then go to next line
    printf("%i\n",coins);
}

File: cash.py
from cs50 import get_float

# reprompt user for change value until it is nonnegative
while True:
    change = get_float("Change: ")
    if change > 0:
        break
# convert dollars to cents
cents = round(change*100)

# diving by 25, outputting the nearest whole number, to get how many quarters to use
coins = cents // 25
# calculate how many cents remain
cents %= 25

# repeat for dimes and nickels, making sure to add onto previous value of coins
coins += cents // 10
cents %= 10
coins += cents // 5

# print total number of coins, which will be the previous number of coins + the remaining change, which is in pennies
print(coins + (cents % 5))